tests/smoke/base.sh

```diff
@@ -49,6 +49,8 @@ elif command -v meson &> /dev/null; then
   pip install -e .
 elif command -v pdm &> /dev/null; then
   pdm install
+elif command -v hatch &> /dev/null; then
+  pip install -e .
```

Comment:
I think this should be `hatch install`? Was that giving any errors?

Reply:
maybe I am wrong, I am not a hatch/hatchling user 
but it seems that hatch doesn't have a command for install just for building
and it seems that for executing it in local mode we should run tests with something like `hatch run pytest .`

so using pip install -e here would be a workaround .. but TBH I don't know if that works
another option would be to build it and install it from the packaged created by `build`
but it probably will not install the dev dependencies (for test and for documentation)

so, not sure yet what could be the best workflow for this to keep it generic ...
otherwise we will need to create an environment variable called for example COMMAND_PREFIX
and add it to all the other commands that will need to have access to the dependencies 
and for hachling we will set it to `hatch run` 

for example:

```bash
COMMAND_PREFIX="hatch run"

$COMMAND_PREFIX pytest
```

but the error on CI looks like some syntax error on pyproject.toml

Reply:
Ah yes, thanks for the clarification!